fix TypeError

federicofantini opened the pull request:
This error is currently breaking [CryptPad 2025.3.1](https://github.com/cryptpad/cryptpad/issues/1921). I tried drafting a fix and it seems to be working now.

federicofantini committed:
fix TypeError

TypeError [ERR_INVALID_ARG_TYPE]: The "paths[1]" argument must be of type string. Received undefined

pirxpilot commented on lib/gzip-static.js:
```diff
@@ -21,17 +21,17 @@ function setHeader(res, path, encoding) {
       return;
     }
-    const { name, parentPath } = dirent;
+    const { name, path: relativePath } = dirent;
```
isn't `path` a deprecated property replaced by `parentPath` - what are we really changing here?

pirxpilot commented:
Thanks for the PR. I am not sure what is the issue that we are trying to fix here. Any chance you could try writing the test that  exposes this bug.

federicofantini commented:
I’ve identified the cause of the issue — I’m running Node.js version `v18.19.0`.

Here’s a quick walkthrough of what I found:

I added the following two lines at [[line 28](https://github.com/pirxpilot/connect-gzip-static/blob/main/lib/gzip-static.js#L28)](https://github.com/pirxpilot/connect-gzip-static/blob/main/lib/gzip-static.js#L28) in `gzip-static.js` to inspect the values:

```js
console.log('Root:', root);
console.log('Dirent:', dirent);
```

The output was:

```
Root: /path/to/cryptpad/www
Dirent: Dirent {
  name: '21B6.png',
  path: '/path/to/cryptpad/www/pad/mathjax/fonts/HTML-CSS/TeX/png/AMS/Regular/085',
  [Symbol(type)]: 1
}
```

This led to the following exception:

```
TypeError [ERR_INVALID_ARG_TYPE]: The "paths[1]" argument must be of type string. Received undefined
    ...
    at path.resolve(...)
    ...
```

The problem is that `dirent` does not contain a `parentPath` field, and `path.resolve(root, parentPath, name)` fails because `parentPath` is undefined.
In my case, the `path` field in the `Dirent` object already contains the directory, so I removed `root` from the concatenation in [[line 34](https://github.com/pirxpilot/connect-gzip-static/blob/main/lib/gzip-static.js#L34)](https://github.com/pirxpilot/connect-gzip-static/blob/main/lib/gzip-static.js#L34), and the error no longer occurs.
